allennlp/training/model.py

```diff
@@ -0,0 +1,64 @@
+from typing import Dict
+import torch
+
+
+class Model(torch.nn.Module):
+    """
+    This abstract class represents a model to be trained. Rather than relying completely
+    on the Pytorch Module, we provide a richer API that maintains more metadata
```

Comment:
Because we're not actually providing more functions, just changing the signature of one of them, I might just say "we modify the `forward` API to make it more suitable for easy training." (and still include your comment about named inputs)